tests/src/test/scala/org/json4s/ExtractionBugs.scala

```diff
@@ -355,9 +355,18 @@ abstract class ExtractionBugs[T](mod: String) extends Specification with JsonMet
       Extraction.extract[OptionOfInt](obj) must_== OptionOfInt(None)
     }
 
+    "Extract should fail when strictOptionParsing is on and extracting from JNull" in {
+      implicit val formats = new DefaultFormats {
+        override val strictOptionParsing: Boolean = true
+      }
+
+      Extraction.extract[OptionOfInt](JNull) must throwA(
+        new MappingException("No value set for Option property: opt")
+      )
+    }
```

Comment:
This difference in error message is because there's explicit handling of `JNull` at the top level of the extraction. I could jump through some hoops in order to not prefix the message with "No usable value for..." but I think a better solution would be to check in [`extract`](https://github.com/json4s/json4s/blob/b77dce3966c56e193eee04d781c0fdd0751daeb1/core/src/main/scala/org/json4s/Extraction.scala#L423) that `json` is a `JObject`. Such a change would probably be better made in a separate PR though.